frontend/src/components/NotificationsBar/NotificationsBarContainer.tsx

```diff
@@ -11,38 +12,40 @@ interface Props {
   notifications?: NotificationType[];
 }
 
-const NotificationsBarContainer: React.FC<Props> = ({
+const NotificationsBarContainer: FC<Props> = ({
   isMobileMenuOpen,
   toggleMobileMenu,
   notifications,
 }) => {
   const [isNotificationsOpen, setNotificationsOpen] = useState<boolean>(false);
-  const [isContainsUnread, setUnread] = useState<boolean>(false);
+
+  const { mutate: readNotificationsMutation } = useReadNotifications();
+
+  const newNotification = notifications?.find(
+    (notification) => notification.status === NOTIFICATION_STATUSES.new,
+  );
 
   const handleNotificationsOpen = () => {
     if (isMobileMenuOpen) {
       toggleMobileMenu();
     }
     setNotificationsOpen(!isNotificationsOpen);
+    if (newNotification) {
+      readNotificationsMutation({});
+    }
   };
 
   const handleNotificationsClose = () => {
     setNotificationsOpen(false);
   };
 
-  useEffect(() => {
-    if (notifications?.find((notification) => notification.status === NOTIFICATION_STATUSES.new)) {
-      setUnread(true);
-    }
-  }, [notifications]);
-
   return (
     <NotificationsBar
       notifications={notifications}
       isNotificationsOpen={isNotificationsOpen}
       handleNotificationsOpen={handleNotificationsOpen}
       handleNotificationsClose={handleNotificationsClose}
-      isContainsUnread={isContainsUnread}
+      isContainsUnread={!!newNotification}
```

Comment:
```suggestion
      isContainsUnread={Boolean(newNotification)}
```

Maybe will be better. More readable